statsmodels/tsa/arima/estimators/hannan_rissanen.py

```diff
@@ -130,15 +133,14 @@ def hannan_rissanen(endog, ar_order=0, ma_order=0, demean=True,
     elif max_ma_order == 0:
         # extract 1) lagged_endog with free params; 2) lagged_endog with fixed
         # params; 3) endog residual after applying fixed params if applicable
-        X_with_free_params = lagged_endog[:, params_info.free_ar_ix]
-        X_with_fixed_params = lagged_endog[:, params_info.fixed_ar_ix]
+        X_with_free_params = lagged_endog[:, free_ar_ix]
+        X_with_fixed_params = lagged_endog[:, fixed_ar_ix]
         y = endog[max_ar_order:]
         if X_with_fixed_params.shape[1] != 0:
-            y = y - X_with_fixed_params.dot(params_info.fixed_ar_params)
+            y = y - X_with_fixed_params.dot(p.fixed_ar_params)
 
         # no free ar params -> variance computation on the endog residual
         if X_with_free_params.shape[1] == 0:
-            p.ar_params = params_info.fixed_ar_params
```

Comment:
No need to set any fixed params again after calling `set_fixed_params` at the beginning once.